### ISSUE TITLE
Easier token revocation

### ISSUE THREAD

**Issue by RoiEXLab** [NodeBB/nodebb-plugin-write-api#124]
Hi, I'm currently trying to implement a system that works for users with 2FA after #121
My current approach tries to simply store the token somewhere on a users system until the user decides to clear it manually.
Simply dropping the token would be easy, but I'd rather have it revoked just to be safe. But in order to revoke it I need to have the uid of the user the token was created for, which I don't really want to store simply because it's not required in order to use the part of the API that I'm using.
Which made me wonder: Is there any specific reason why the uid is required?
It makes sense from the endpoint scheme, revocation is just right next to generation.
But wouldn't it be easier to just have an endpoint that simply revokes the currently used token?

**Comment by julianlam** [comment]
The whole token revocation logic does need a little bit of rethinking.

In NodeBB version 2, the write API will be merged into core. At that time, it does make more sense for the token generation and revocation routes to be in their own namespace, as opposed to the users namespace. To that end, there wouldn't need to be the requirement to pass in the uid, but for the write API as it is now, it's a bit of a breaking change, so I don't think I want to add it right now, does that make sense?

**Comment by RoiEXLab** [comment]
Sure, no problem.
Just wanted to bring up this topic for consideration

**Comment by julianlam** [comment]
Will address in v2 :smile:

**Comment by RoiEXLab** [comment]
@julianlam Good to hear 👍 

Just out of curiosity: Is there some sort of fixed milestone for v2?
I just saw the webpack branch so I assume v2 will be a big refactoring change?

**Comment by julianlam** [comment]
It'll be huge :grimacing: -- but for the Write API, it'll be brought into core and most of the routes will remain the same.

There will be a migration doc :smile:
